cdc/generation.cc

```diff
@@ -875,24 +895,8 @@ future<> generation_service::check_and_repair_cdc_streams() {
                 " even though some node gossiped about it.",
                 latest, db_clock::now());
             should_regenerate = true;
-        } else {
-            if (tmptr->sorted_tokens().size() != gen->entries().size()) {
-                // We probably have garbage streams from old generations
-                cdc_log.info("Generation size does not match the token ring, regenerating");
-                should_regenerate = true;
-            } else {
-                std::unordered_set<dht::token> gen_ends;
-                for (const auto& entry : gen->entries()) {
-                    gen_ends.insert(entry.token_range_end);
-                }
-                for (const auto& metadata_token : tmptr->sorted_tokens()) {
-                    if (!gen_ends.contains(metadata_token)) {
-                        cdc_log.warn("CDC generation {} missing token {}. Regenerating.", latest, metadata_token);
-                        should_regenerate = true;
-                        break;
-                    }
-                }
-            }
+        } else if (is_cdc_generation_optimal(*gen, *tmptr)) {
```

Comment:
I don't think this is right.

I think a new version will be needed after all...

Reply:
I have negated the condition. It should be fine now.